vaadin-map-flow-parent/vaadin-map-flow/src/main/java/com/vaadin/flow/component/map/configuration/AbstractConfigurationObject.java

```diff
@@ -84,4 +119,13 @@ public void update(Runnable updater, boolean notifyChanges) {
             this.notifyChanges.remove();
         }
     }
+
+    public void collectChanges(
+            Consumer<AbstractConfigurationObject> changeCollector) {
+        children.forEach(child -> child.collectChanges(changeCollector));
```

Comment:
Ensures that nested references are synced first, so that they are available when synchronizing this instance.